src/engine/ITableau.h

```diff
@@ -180,6 +180,16 @@ class ITableau
     virtual bool areLinearlyDependent( unsigned x1, unsigned x2, double &coefficient, double &inverseCoefficient ) = 0;
     virtual unsigned getVariableAfterMerging( unsigned variable ) const = 0;
     virtual void postContextPopHook() = 0;
+
+    bool isOptimizing() const
+    {
+        return _optimizing;
+    }
+    void optimizing() { _optimizing = true; }
```

Comment:
This is a bit inconsistent with the rest of the code. Can we switch to a single method, e.g., toggleOptimization( bool ), instead of having two setters?

Reply:
Done